service/src/main/kotlin/fi/espoo/evaka/holidayperiod/HolidayPeriodControllerCitizen.kt

```diff
@@ -48,7 +51,18 @@ class HolidayPeriodControllerCitizen(private val accessControl: AccessControl) {
 
         db.connect { dbc ->
             dbc.transaction { tx ->
+                val activeHolidayPeriod = tx.getActiveFreeHolidayPeriod(evakaClock.today())
+                if (activeHolidayPeriod?.freePeriod == null) {
+                    throw NotFound("No active free holiday period found")
+                }
+
                 val childToDateRange = body.childHolidays.entries.filter { it.value != null }.map { it.key to it.value!! }
+                childToDateRange.forEach { (_, dateRange) ->
+                    if (!activeHolidayPeriod.freePeriod.periodOptions.contains(dateRange)) {
+                        throw NotFound("Free holiday period not found")
+                    }
+                }
```

Comment:
Using `also` would be slightly more idiomatic Kotlin. Moreover, how about `BadRequest` in both of the `throw` cases instead of `NotFound`?